src/Compatibility/ApiCompat/Microsoft.DotNet.PackageValidation/ApiCompatRunnerExtensions.cs

```diff
@@ -31,6 +32,19 @@ internal static class ApiCompatRunnerExtensions
                 return;
             }
 
+            // Don't perform api compatibility checks on placeholder files.
+            if (leftContentItems.IsPlaceholderFile() && rightContentItems.IsPlaceholderFile())
+            {
+                leftContentItems[0].Properties.TryGetValue("tfm", out object? tfmRaw);
+                string tfm = tfmRaw?.ToString() ?? string.Empty;
+
+                log.LogMessage(MessageImportance.Normal, string.Format(Resources.SkipApiCompatForPlaceholderFiles, tfm));
+            }
+
+            // Make sure placeholder package files aren't enqueued as api comparison check work items.
+            Debug.Assert(!leftContentItems.IsPlaceholderFile() && !rightContentItems.IsPlaceholderFile(),
```

Comment:
I don't think this is an accurate assert.  What if a package explicitly adds a placeholder in a compatible framework?

It might do this if a) the library became part of that framework and we could compare the package library to the framework library (in references) or b) as an intentional breaking change in which case we should still compare and raise the missing file.